cpp/ql/src/semmle/code/cpp/Class.qll

```diff
@@ -857,8 +892,16 @@ class AbstractClass extends Class {
 }
 
 /**
- * A class template. (This class also finds partial specializations
- * of class templates).
+ * A class template (this class also finds partial specializations
+ * of class templates).  For example in the following code there is a
+ * `MyTemplateClass<T>` template:
```

Comment:
I think it is worth (a) adding an example of a partial specialization, and (b) adding an example of a full specialization and an instantiation and saying they are not included, but linking to the classes that do include them.

Reply:
I'm reluctant to add all of those as examples since this would lead to an extremely long block of help.  I've added a reference to `ClassTemplateInstantiation` and `FullClassTemplateSpecialization`.